File: client/iih_stub/iih.ci_stub/iih.ci.mr/nu/generalnursingrec/d/GeneralNursingRecDO.cs

using System;
using xap.mw.core.data;
using xap.mw.coreitf.d;
using System.Collections.Generic;

namespace iih.ci.mr.nu.generalnursingrec.d
{
    /// <summary>
    /// GeneralNursingRecDO 的摘要说明。
    /// </summary>
    public class GeneralNursingRecDO : BaseDO {

		public const string TABLE_NAME = "CI_MR_NU_GNR_REC";
		public const string TABLE_ALIAS_NAME = "a1";

        public GeneralNursingRecDO() {
        }
		public string Id_gnr_rec {
            get { return getAttrVal<string>("Id_gnr_rec",null); }
            set { setAttrVal<string>("Id_gnr_rec", value); }
        }
		public string Id_gnr {
            get { return getAttrVal<string>("Id_gnr",null); }
            set { setAttrVal<string>("Id_gnr", value); }
        }
		public string Conditiontreatment {
            get { return getAttrVal<string>("Conditiontreatment",null); }
            set { setAttrVal<string>("Conditiontreatment", value); }
        }
		public DateTime? Dt_rec {
            get { return getAttrVal<FDateTime>("Dt_rec",null); }
            set { setAttrVal<FDateTime>("Dt_rec", value); }
        }
		public string Signature {
            get { return getAttrVal<string>("Signature",null); }
            set { setAttrVal<string>("Signature", value); }
        }
		public string Createdby {
            get { return getAttrVal<string>("Createdby",null); }
            set { setAttrVal<string>("Createdby", value); }
        }
		public DateTime? Createdtime {
            get { return getAttrVal<FDateTime>("Createdtime",null); }
            set { setAttrVal<FDateTime>("Createdtime", value); }
        }
		public string Modifiedby {
            get { return getAttrVal<string>("Modifiedby",null); }
            set { setAttrVal<string>("Modifiedby", value); }
        }
		public DateTime? Modifiedtime {
            get { return getAttrVal<FDateTime>("Modifiedtime",null); }
            set { setAttrVal<FDateTime>("Modifiedtime", value); }
        }
		public string Name_sign {
            get { return getAttrVal<string>("Name_sign",null); }
            set { setAttrVal<string>("Name_sign", value); }
        }
        public int Ds {
            get { return getAttrVal<int>("Ds",0);}
            set { setAttrVal<int>("Ds", value); }
        }

        public DateTime? Sv        {
            get { return getAttrVal<FDateTime>("Sv",null); }
            set { setAttrVal<FDateTime>("Sv", value); }
        }
        
        /// <summary>
        /// 返回表名
        /// </summary>
        /// <returns></returns>
        public override string getTableName()
        {
            return "CI_MR_NU_GNR_REC";
        }
        
        /// <summary>
        /// 返回表别名
        /// </summary>
        /// <returns></returns>
        public override string getAliasTableName()
        {
            return "a1";
        }
        
        /// <summary>
        /// 返回主键字段名
        /// </summary>
        /// <returns></returns>
        public override string getPKFieldName()
        {
            return "Id_gnr_rec";
        }

        /// <summary>
        /// 返回全路径DO类名
        /// </summary>
        /// <returns></returns>
        public override string getFullClassName()
        {
            return "iih.ci.mr.nu.generalnursingrec.d.GeneralNursingRecDO";
        }

        /// <summary>
        /// 返回属性列表
        /// </summary>
        /// <returns></returns>
        public override string[] getAttrNames()
        {
            return new string[]{"Ds","Sv",  "Id_gnr_rec", "Id_gnr", "Conditiontreatment", "Dt_rec", "Signature", "Createdby", "Createdtime", "Modifiedby", "Modifiedtime", "Name_sign"};
        }
        
    }
}

File: server/iih_pub/iih.ci_pub/src/main/java/iih/ci/mr/nu/generalnursingrec/d/GeneralNursingRecDO.java
package iih.ci.mr.nu.generalnursingrec.d;

import xap.sys.appfw.orm.desc.DescManager;
import xap.mw.coreitf.c.GlobalConst;
import xap.mw.coreitf.i.IDODesc;
import xap.mw.coreitf.d.*;
import xap.mw.core.data.*;
import iih.ci.mr.nu.generalnursingrec.d.desc.GeneralNursingRecDODesc;
import java.math.BigDecimal;

/**
 * 一般护理记录 DO数据 
 * 
 */
public class GeneralNursingRecDO extends BaseDO {
	private static final long serialVersionUID = 1L;

	public static final String ID_GNR_REC= "Id_gnr_rec";
	public static final String ID_GNR= "Id_gnr";
	public static final String CONDITIONTREATMENT= "Conditiontreatment";
	public static final String DT_REC= "Dt_rec";
	public static final String SIGNATURE= "Signature";
	public static final String CREATEDBY= "Createdby";
	public static final String CREATEDTIME= "Createdtime";
	public static final String MODIFIEDBY= "Modifiedby";
	public static final String MODIFIEDTIME= "Modifiedtime";
	public static final String NAME_SIGN= "Name_sign";
	public static final String DS = "Ds";
	public static final String SV = "Sv";
	
	public String getId_gnr_rec() {
		return ((String) getAttrVal("Id_gnr_rec"));
	}	
	public void setId_gnr_rec(String Id_gnr_rec) {
		setAttrVal("Id_gnr_rec", Id_gnr_rec);
	}
	public String getId_gnr() {
		return ((String) getAttrVal("Id_gnr"));
	}	
	public void setId_gnr(String Id_gnr) {
		setAttrVal("Id_gnr", Id_gnr);
	}
	public String getConditiontreatment() {
		return ((String) getAttrVal("Conditiontreatment"));
	}	
	public void setConditiontreatment(String Conditiontreatment) {
		setAttrVal("Conditiontreatment", Conditiontreatment);
	}
	public FDateTime getDt_rec() {
		return ((FDateTime) getAttrVal("Dt_rec"));
	}	
	public void setDt_rec(FDateTime Dt_rec) {
		setAttrVal("Dt_rec", Dt_rec);
	}
	public String getSignature() {
		return ((String) getAttrVal("Signature"));
	}	
	public void setSignature(String Signature) {
		setAttrVal("Signature", Signature);
	}
	public String getCreatedby() {
		return ((String) getAttrVal("Createdby"));
	}	
	public void setCreatedby(String Createdby) {
		setAttrVal("Createdby", Createdby);
	}
	public FDateTime getCreatedtime() {
		return ((FDateTime) getAttrVal("Createdtime"));
	}	
	public void setCreatedtime(FDateTime Createdtime) {
		setAttrVal("Createdtime", Createdtime);
	}
	public String getModifiedby() {
		return ((String) getAttrVal("Modifiedby"));
	}	
	public void setModifiedby(String Modifiedby) {
		setAttrVal("Modifiedby", Modifiedby);
	}
	public FDateTime getModifiedtime() {
		return ((FDateTime) getAttrVal("Modifiedtime"));
	}	
	public void setModifiedtime(FDateTime Modifiedtime) {
		setAttrVal("Modifiedtime", Modifiedtime);
	}
	public String getName_sign() {
		return ((String) getAttrVal("Name_sign"));
	}	
	public void setName_sign(String Name_sign) {
		setAttrVal("Name_sign", Name_sign);
	}

	public Integer getDs() {
		return ((Integer) getAttrVal(GlobalConst.DELETE_SIGN));
	}
	
	public void setDs(Integer ds) {
		setAttrVal(GlobalConst.DELETE_SIGN, ds);
	}
	
	public FDateTime getSv() {
		return ((FDateTime) getAttrVal(GlobalConst.SYSTEM_VERSION));
	}
	
	public void setSv(FDateTime sv) {
		setAttrVal(GlobalConst.SYSTEM_VERSION, sv);
	}

//	@Override
//	public java.lang.String getParentPKFieldName() {
//		return null;
//	}
  
	@Override
	public String getPKFieldName() {
		return "Id_gnr_rec";
	}
	
	@Override
	public String getTableName() {	  
		return "CI_MR_NU_GNR_REC";
	}
	
	@Override
	public IDODesc getDODesc() {
		return DescManager.getInstance().getDODesc(GeneralNursingRecDODesc.class);
	}
	
}